### PR TITLE
Feature external audio option

### PR DESCRIPTION
**Pull request by apotropaic**
Still missing the logic for the feed here. Will add that later

### CODE REVIEW

File: templates/podcast-episode.html.twig [hunk 1]
```diff
@@ -7,7 +7,12 @@
     {% endif %}
     <h3>Posted {{ header.date|date('Y-m-d')|nicetime(false) }}</h3>
     {{ page.content }}
-    {{ page.media[ (page.header.podcast.audio|first).name ].html() }}
+    {% if page.header.podcast.audio %}
+        {% set audio_file = page.header.podcast.audio|first.name  %}
+    {% elseif page.header.podcast.audio %}
```
Comment (jgonyea) [inline templates/podcast-episode.html.twig]: This condition is the same as L10 above?

File: podcast.php [hunk 2]
```diff
@@ -122,6 +156,48 @@ public static function retreiveAudioDuration($file)
...
+            {
+                while ($buffer = fread($fp_remote, 8192)) 
+                {
+                    fwrite($fp_local, $buffer);
+                }
+                fclose($fp_local);
+
+                return $localtempfilename;
+            }
+            fclose($fp_remote);
```
Comment (jgonyea) [inline podcast.php]: This is never reached if the function returns $localtempfilename.

File: podcast.php [hunk 3]
```diff
@@ -122,6 +156,48 @@ public static function retreiveAudioDuration($file)
...
+     *     Audio filesize.
+     */
+    public static function getRemoteAudio($url)
+    {
+        if ($fp_remote = fopen($url, 'rb')) 
+        {
+            $localtempfilename = tempnam('/tmp', 'getID3');
+            if ($fp_local = fopen($localtempfilename, 'wb'))
+            {
+                while ($buffer = fread($fp_remote, 8192))
```
Comment (jgonyea) [inline podcast.php]: Example 3 on this page: http://php.net/manual/en/function.fread.php
has some better remote file handling.

Comment (apotropaic) [reply podcast.php]: agreed, I grabbed this one from the getid3 php website - I should have looked a little closer at it. I can redo this for sure and make it better base don that exampel.

File: templates/podcast-episode.html.twig (hunk 1 above)
Comment (jgonyea) [reply templates/podcast-episode.html.twig]: if page.header.podcast.audio
vs
elseif page.header.podcast.audio

**Comment by jgonyea** [comment]
first impressions:
I like the concept!  I'm trying to rework the metadata saving using onAdminSave, which will probably change your PR slightly.

File: templates/podcast-episode.html.twig (hunk 1 above)
Comment (apotropaic) [reply templates/podcast-episode.html.twig]: Ooh... yea. I see what  you are saying. I think its supposed to say `page.header.podcast.audio_url`

**Comment by apotropaic** [comment]
Yes onAdminSave would change things a little, but it should be quite a bit more efficient to go that route

**Comment by apotropaic** [comment]
Got these changes fixes

**Comment by jgonyea** [comment]
Due to my need for some underlying changes, I'm rejecting this particular PR due to incompatibilities. Since I ended up using a lot of code from this PR, you'll be included as a contributor in v2.1.0.

**Pull request closed without merging**